### DISCUSSION TITLE
Navigation in generated .swiftinterface files

### DISCUSSION THREAD

**Discussion by akaralar** [wojciech-kulik/xcodebuild.nvim#122] [General]
Is there a way to make the LSP integration work with `.swiftinterface` files? For instance when I go to definition of a `SwiftUI` symbol, say `Text`, I am directed to a generated `SwiftUI.swiftinterface` file located in a predefined path which can be seen by running 
```
xcrun --toolchain swift sourcekit-lsp --help
```

Later in this `SwiftUI.swiftinterface` file, if I try to go to definition of a symbol in another `Swift` module, say `String`, it doesn't navigate there and just finds occurrences in the current file. Even though I change the `filetype` of a `.swiftinterface` file in Neovim to be a `swift` file (so that tree-sitter works correctly), I don't see the LSP attached when I check with `:LspInfo`. 

If I go back to my project files and try to navigate to `String` from one of the project files, it navigates me to a `Swift.String.swiftinterface` file in that same folder for generated interfaces. So I'm guessing there should be a way to send an LSP request for generating an interface even when the originating file is in that generated interfaces folder. 

I'm not sure if this issue is better suited to the `xcode-build-server` repo or this one but since I started to interface with LSP through this repo, I wanted to open an issue here first.

**Comment by wojciech-kulik** [comment]
Have you tried setting "sourcekit-lsp" for `swiftinterface` format? If you set it for just `swift`, it won't automatically attach to `swiftinterface` buffer.

**Comment by wojciech-kulik** [comment]
```lua
    lspconfig["sourcekit"].setup({
      capabilities = capabilities,
      on_attach = on_attach,
      filetypes = { "swift", "swiftinterface", "c", "cpp", "objective-c", "objective-cpp" },
    })
```

**Comment by akaralar** [comment]
I have just tried it and it doesn't make a difference. this is what I see on `:LspInfo`

![image](https://github.com/wojciech-kulik/xcodebuild.nvim/assets/309655/8c301c01-2973-427c-bb8e-93e7c8078257)

**Comment by wojciech-kulik** [comment]
I'm not familiar with sourcekit-lsp internals, I think it would be best to create a ticket here: https://github.com/apple/sourcekit-lsp

**Comment by Woit** [comment]
@akaralar try to add this to yours autocommands

```lua
-- append filetype .swiftinterface to swift group
vim.api.nvim_create_autocmd({ "BufRead", "BufNewFile" }, {
  group = augroup("xcode_swiftinterface"),
  pattern = { "*.swiftinterface" },
  callback = function()
    vim.bo.ft = "swift"
  end,
})
```

**Comment by akaralar** [comment]
@Woit I replaced your `augroup("xcode_swiftinterface")` with `vim.api.nvim_create_augroup("xcode_swiftinterface", { })` and it doesn't make a difference, LSP is not attached. 

To make things clear, I already set filetype of `.swiftinterface` files to `swift` using `ftdetect` mechanism, I have `swiftinterface.vim` file in `~/.config/nvim/ftdetect` that says
```
au BufRead,BufNewFile *.swiftinterface                setfiletype swift
```

so the reason the LSP is not attaching is not because of the filetype. Does LSP in those `.swiftinterface` files work for you?

**Comment by Woit** [comment]
@akaralar you're right, lsp is not attached to .swiftframwork

**Comment by wojciech-kulik** [comment]
I converted this issue into the discussion, as it is not directly related to the plugin functionality.

**Comment by akaralar** [comment]
@Woit I managed to get it to attach by enabling single file mode, ie 
```lua
lspconfig["sourcekit"].setup({
  single_file_support = true, -- Added this line to the config
  capabilities = capabilities,
  on_attach =  on_attach
})
```
But still navigation does not work. 

I also opened https://github.com/apple/sourcekit-lsp/issues/1155 on the sourcekit repo to see if anyone has any ideas there

**Comment by wojciech-kulik** [comment]
I'm closing discussion, because now it's supported by default.
